doc/static-route/SONiC_static_route_bfd_hld.md

```diff
@@ -0,0 +1,300 @@
+# Scope
+
+This document describes the design of using BFD to detect static route nexthop and based on BFD session state, installing, updating, and removing static route to the system.<br  /> 
+
+One important assumption is that the BFD session (for a static route nexthop) in this BFD Static Route application is not shared with any other applications (i.e., creating or deleting static route BFD session has no impact on other applications).<br /> 
+
+<span style="color:blue"> <br>
+*Note: This document describes how to create local BFD session for a static route based on static route configuration. Creating BFD session in its peer system is out of the scope of this document.*<br />  
+</span>
+
+
+# Static Route BFD Design
+## Functional Requirements
+
+1. Create BFD session for each static route nexthop based on static route configuration <br>
+2. Delete BFD session if the nexthop is not needed (triggered by removing static route from configration)<br>
+3. Install, update, removing static route (i.e., update nexthop, remove static route if no nexthop) based on BFD session state change.<br>
+4. When the application (static route BFD) restarts, recover static routes and BFD session states from redis database (config_db, appl_db and state_db) without impacting any existing installed static routes and BFD sessions. <br  />  
+
+## System Overview with Static Route BFD
```

Comment:
I think, it makes sense to introduce new staticroutebfd as a separate manager to handle static route with bfd configs from config-db. This will help to make use of this feature for static route configuration based on frrcfgd as well not just bgpcfgd whenever the user doesnot use bgpcfgd.

Reply:
the problem is that, the manage in runner/manager framework monitors one table only. but staticroutebfd needs to monitor two tables at same time: static route table and bfd session state table. The current manager cannot meet this requirement.